internal/cmd/asyncapi/command_import.go

```diff
@@ -0,0 +1,391 @@
+	updateReq := srsdk.ConfigUpdateRequest{Compatibility: compatibility}
+	config, _, err := details.srClient.DefaultApi.UpdateSubjectLevelConfig(details.srContext, subject, updateReq)
+	if err != nil {
+		return fmt.Errorf("failed to update subject level compatibility: %v", err)
+	}
+	log.CliLogger.Infof("subject level compatibility updated to %s", config.Compatibility)
+	return nil
+}
+
+func addSchemaTags(details *accountDetails, components Components, topicName string, schemaId int32) ([]srsdk.Tag, []srsdk.TagDef, error) {
+	//Schema level tags
+	var tagConfigs []srsdk.Tag
+	var tagDefConfigs []srsdk.TagDef
+	if components.Messages != nil {
+		for _, tag := range components.Messages[strcase.ToCamel(topicName)+"Message"].Tags {
+			tagDefConfigs = append(tagDefConfigs, srsdk.TagDef{
+				EntityTypes: []string{"cf_entity"},
+				Name:        tag.Name,
+			})
+			tagConfigs = append(tagConfigs, srsdk.Tag{
+				TypeName:   tag.Name,
+				EntityType: "sr_schema",
+				EntityName: details.srCluster.Id + ":.:" + strconv.Itoa(int(schemaId)),
+			})
+		}
+		err := addTagsUtil(details, tagDefConfigs, tagConfigs)
+		return tagConfigs, tagDefConfigs, err
+	}
+	return tagConfigs, tagDefConfigs, nil
+}
+
+func addTopicTags(details *accountDetails, subscribe Operation, topicName string) ([]srsdk.Tag, []srsdk.TagDef, error) {
+	//Topic level tags
+	//add topic level tags only if topic was created successfully or already exists
+	var tagConfigs []srsdk.Tag
+	var tagDefConfigs []srsdk.TagDef
+	for _, tag := range subscribe.TopicTags {
+		tagDefConfigs = append(tagDefConfigs, srsdk.TagDef{
+			EntityTypes: []string{"cf_entity"},
+			Name:        tag.Name,
+		})
+		tagConfigs = append(tagConfigs, srsdk.Tag{
+			TypeName:   tag.Name,
+			EntityType: "kafka_topic",
+			EntityName: details.srCluster.Id + ":" + details.cluster.Id + ":" + topicName,
+		})
+	}
+	err := addTagsUtil(details, tagDefConfigs, tagConfigs)
+	return tagConfigs, tagDefConfigs, err
+}
+
+func addTagsUtil(details *accountDetails, tagDefConfigs []srsdk.TagDef, tagConfigs []srsdk.Tag) error {
+	tagDefOpts := new(srsdk.CreateTagDefsOpts)
+	tagDefOpts.TagDef = optional.NewInterface(tagDefConfigs)
+	defs, _, err := details.srClient.DefaultApi.CreateTagDefs(details.srContext, tagDefOpts)
+	if err != nil {
+		return fmt.Errorf("unable to create tag definition: %v", err)
+	}
+	log.CliLogger.Debugf("Tag Definitions created: %v", defs)
+	tagOpts := new(srsdk.CreateTagsOpts)
+	tagOpts.Tag = optional.NewInterface(tagConfigs)
+	tags, _, err := details.srClient.DefaultApi.CreateTags(details.srContext, tagOpts)
+	log.CliLogger.Debugf("%v added to resource %s", tags, tagConfigs[0].EntityName)
+	if err != nil {
+		return fmt.Errorf("unable to add tag to resource: %v", err)
+	}
+	log.CliLogger.Infof("%v added to resource %s", tags, tagConfigs[0].EntityName)
+	return nil
+}
+
+//Add Topics
+//Register a schema against that subject
+//Tags at Schema Level
+//Tags at topic Level
```

Comment:
Are these tasks that you still need to do?

Reply:
No, this is just the workflow being done by the command.